sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,288 @@
+
+
+def parse_times(table):
+    """
+    Parses the time values in an Astropy table into `astropy.time.Time`.
+
+    Parameters
+    ----------
+    table : `astropy.table.Table`
+        Astropy table.
+
+    Returns
+    -------
+    `astropy.table.Table`
+    """
+    # All time columns from https://www.lmsal.com/hek/VOEvent_Spec.html
+    time_keys = ['event_endtime', 'event_starttime', 'event_peaktime']
+    for tkey in time_keys:
+        if tkey in table.colnames:
+            table[tkey] = [
+                (parse_time(time, format='iso') if time else numpy.ma.masked) for time in table[tkey]
+            ]
+    return table
+
+
+def parse_values_to_quantities(table):
+    """
+    Parses the values in an Astropy table into Astropy objects.
+
+    Parameters
+    ----------
+    table : `astropy.table.Table`
+        Astropy table.
+
+    Returns
+    -------
+    `astropy.table.Table`
+
+    Raises
+    ------
+    TypeError
+        If ``table`` is not an Astropy table.
+    """
+    with open(UNIT_FILE_PATH) as unit_file:
+        unit_properties = json.load(unit_file)
+    unit_attributes = unit_properties["attributes"]
+
+    with open(COORD_FILE_PATH) as coord_file:
+        coord_properties = json.load(coord_file)
+    coord_attributes = coord_properties["attributes"]
+    table = parse_columns_to_table(table, unit_attributes)
+    table = parse_columns_to_table(table, coord_attributes, is_coord_prop = True)
+    return table
+
+def parse_columns_to_table(table, attributes, is_coord_prop = False):
+    """
+    Parses the columns in an Astropy table and convert the values into Astropy objects.
+
+    Parameters
+    ----------
+    table: `astropy.table.Table`
+        Astropy table.
+    attributes : list
+        A list of HEK unit attributes or coordinate attributes.
+    is_coord_prop : bool
+        To specify if ``attributes`` is a list of unit attributes or coordinate attributes.
+
+    Returns
+    -------
+    `astropy.table.Table`
+
+    Raises
+    ------
+    TypeError
+        If ``table`` is not an Astropy table.
+    KeyError
+        If any of the attribute dictionaries are missing required keys (i.e. "name", "unit_prop").
+    """
+    if is_coord_prop:
+        event_coord_col = []
+        for idx in range(len(table['event_coord1'])):
+            unit = get_unit(table['event_coordunit'][idx])
+            coord1 = table['event_coord1'][idx]
+            coord2 = table['event_coord2'][idx]
+            coord3 = table['event_coord3'][idx]
+            time = table['event_starttime'][idx]
+            frame = 'helioprojective' if unit==u.arcsec else 'icrs'
+            if coord3:
+                event_coord = SkyCoord(coord1*unit, coord2*unit, coord3*unit, obstime=time, frame=frame)
+            else:
+                event_coord = SkyCoord(coord1*unit, coord2*unit, obstime=time, frame=frame)
+            event_coord_col.append(event_coord)
+        event_coord_col = Column(event_coord_col, name='event_coord')
```

Comment:
We should be building a single `SkyCoord` object here and using it as a column rather than a Column of individual SkyCoords. It's possible to make a SkyCoord with a observer attribute which itself is array like (with the same shape as the data)

Reply:
Oftentimes many of the entries will be masked. I wonder whether it is possible to construct a single multidimensional SkyCoord in a masked column?

Reply:
Ok so masking isn't too much of an issue In astropy>=7.0 and I think I've now got that working. As far as casting to a single coordinate goes, I have this now working in all cases where the data in each row of the column has the same shape (even if it is multidimensional). However, there are a few columns (`bound_chaincode` and `<coord_name>_boundcc` columns) where the data changes shape from row to row. I think this is because each one of these rows represents the boundary of a different feature as defined by some arbitrary number of points (as opposed to the bounding box which is always defined by 5). Unless I'm missing something, there is no way to avoid representing these columns as lists of `SkyCoords` with `dtype='object'`.